Emby.Server.Implementations/Session/WebSocketController.cs

```diff
@@ -57,7 +80,17 @@ private async void OnConnectionClosed(object? sender, EventArgs e)
         {
             var connection = sender as IWebSocketConnection ?? throw new ArgumentException($"{nameof(sender)} is not of type {nameof(IWebSocketConnection)}", nameof(sender));
             _logger.LogDebug("Removing websocket from session {Session}", _session.Id);
-            _sockets.Remove(connection);
+            ObjectDisposedException.ThrowIf(_disposed, this);
+            try
+            {
+                _socketsLock.EnterWriteLock();
+                _sockets.Remove(connection);
+            }
+            finally
+            {
+                _socketsLock.ExitWriteLock();
+            }
+
             connection.Closed -= OnConnectionClosed;
```

Comment:
I might lock the mutations of each connections as well because the handler might be called multiple times. But this is probably fine for this handler.

Reply:
i mean we have the scope, why not include it anway.